examples/common/avb.c

```diff
@@ -87,47 +87,72 @@ int pci_connect(device_t *igb_dev)
 	return 0;
 }
 
-int gptpinit(int *shm_fd, char **memory_offset_buffer)
+int gptpinit(int *igb_shm_fd, char **igb_mmap)
 {
-	*shm_fd = shm_open(SHM_NAME, O_RDWR, 0);
-	if (*shm_fd == -1) {
+	if (igb_shm_fd == NULL || igb_mmap == NULL) {
+		return -1;
+	}
+	*igb_shm_fd = shm_open(SHM_NAME, O_RDWR, 0);
+	if (*igb_shm_fd == -1) {
 		perror("shm_open()");
-		return false;
+		return -1;
 	}
-	*memory_offset_buffer =
-	    (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE, MAP_SHARED,
-			 *shm_fd, 0);
-	if (*memory_offset_buffer == (char *)-1) {
+	*igb_mmap = (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE,
+				MAP_SHARED, *igb_shm_fd, 0);
+	if (*igb_mmap == (char *)-1) {
 		perror("mmap()");
-		*memory_offset_buffer = NULL;
+		*igb_mmap = NULL;
 		shm_unlink(SHM_NAME);
-		return false;
+		return -1;
 	}
-	return true;
+	return 0;
 }
 
-void gptpdeinit(int shm_fd, char *memory_offset_buffer)
+/* gptpdeinit
+ * Return values:
+ *  0 = Success
+ * -1 = close failed
+ * -2 = munmap failed
+ * -3 = close and munmap failed
+ * */
+
+int gptpdeinit(int *igb_shm_fd, char **igb_mmap)
 {
-	if (memory_offset_buffer != NULL) {
-		munmap(memory_offset_buffer, SHM_SIZE);
+	int ret = 0;
+	if (igb_shm_fd == NULL) {
+		ret -= 1;
+	}else if (*igb_shm_fd != -1) {
+		if(close(*igb_shm_fd) == -1) {
+			ret -= 1;
+		}
+		*igb_shm_fd = -1;
 	}
-	if (shm_fd != -1) {
-		close(shm_fd);
+	if (igb_mmap == NULL) {
```

Comment:
igb_mmap should be 0 if not allocated, and if the close above fails for whatever unlikely reason igb_mmap won't be unmapped. (shm_map now)